cpp/src/Slice/SliceUtil.cpp

```diff
@@ -483,7 +483,15 @@ Slice::checkIdentifier(const string& id)
         name = id;
     }
 
+    // Check that the identifier isn't empty.
+    if (name.empty())
```

Comment:
Is it possible to reach this code?

If yes, please add a test.
If not, an assert would be more appropriate.

Reply:
After some research, it should be impossible to hit.
So, I switched it to an assert.

I hit this by accident when I was using it to check modules, and saw that passing in `""` gets a warning of "illegal trailing underscore"...... which is obviously not true (It's a fluke of how we check for trailing underscores).